GoogleForCreators/web-stories-wp
# Design System: Snackbar Accessibility

## Context

Our current snackbars are not really accessible, as they are not announced to screen readers, and it's impossible for users to interact with them.

## Summary

- Use `useLiveRegion` to announce the snackbar messages to users
- `dissmissable` was spelled wrong, so that's fixed here too

## Relevant Technical Choices

- Hiding the snackbar from screen readers for now. If we use `useLiveRegion` then the notification is announced twice. 
- Added a new prop: `actionHelpText`. This is text that is only announced via screen reader. It is not rendered to the page.

## To-do

n/a

## User-facing changes

Nothing visual

## Testing Instructions






This PR can be tested by following these steps:

1. Use a screen reader to test this change. An option in chrome is the [chromevox chrome plugin](https://chrome.google.com/webstore/detail/screen-reader/kgejglhpjiefppelpmljglcjbhoiplfn)
2. Perform an action that shows a snackbar (ex: right click an element to open the context menu. Then click 'copy styles' or something).
3. The snackbar text should have been announced. If there is an action, there is helper text with that action.

## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist




---



Fixes #6061
